presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/GeoFunctions.java

```diff
@@ -525,6 +527,30 @@ public static Slice stExteriorRing(@SqlType(GEOMETRY_TYPE_NAME) Slice input)
     @SqlType(GEOMETRY_TYPE_NAME)
     public static Slice stIntersection(@SqlType(GEOMETRY_TYPE_NAME) Slice left, @SqlType(GEOMETRY_TYPE_NAME) Slice right)
     {
+        if (deserializeType(left) == ENVELOPE && deserializeType(right) == ENVELOPE) {
```

Comment:
Just curious. How common are the situations expressed in this branch?

Reply:
This PR came out of investigation into slow query which was doing 3B intersections of envelopes. The goal of that query was to normalize a dataset gridded one way into bing tiles. The original dataset has (value, cell) columns. The new dataset has (value, bing_tile) columns. To convert, each row is mapped to a set of bing tiles covering the cell, the a weight is assigned to each tile based on the area of the intersection between the tile and the cell, and finally the value is split between tiles according to the weight.

`ST_Area(ST_Intersection(cell, bing_tile_polygon(tile)) / ST_Area(cell) as weight`